OMX-27-firmware/OMX-27-firmware.ino

```diff
@@ -542,75 +592,86 @@ void show_current_step(int patternNum) {
 			}
 		}
 
-		for(int i = 0; i < NUM_STEPKEYS; i++){
-			if (i < PatternLength(patternNum)){
+		
+		auto currentpage = patternPage[patternNum];
+		auto pagestepstart = (currentpage * NUM_STEPKEYS);
+		
+												// WHAT TO DO HERE FOR MULTIPLE PAGES
+		for(int i = pagestepstart; i < (pagestepstart + NUM_STEPKEYS); i++){	// NUM_STEPKEYS or NUM_STEPS INSTEAD?>
```

Comment:
```suggestion
		// WHAT TO DO HERE FOR MULTIPLE PAGES
		// NUM_STEPKEYS or NUM_STEPS INSTEAD?
		for(int i = pagestepstart; i < (pagestepstart + NUM_STEPKEYS); i++){
```

Reply:
also... maybe something like this
```suggestion
		// WHAT TO DO HERE FOR MULTIPLE PAGES
		// NUM_STEPKEYS or NUM_STEPS INSTEAD?
		for(int i = pagestepstart; i < PatternLength(patternNum) && i < (pagestepstart + NUM_STEPKEYS); i++){
```

Reply:
This whole for loop needs to be rewritten, there's too much duplicate code for me to follow right now.